### ISSUE TITLE
Unable to do set command

### ISSUE THREAD

**Issue by nicolouts** [Alexandre-io/homeassistant-vcontrol#13]
Hello,
I have follow the configuration from Bjoern3003.
My device protocol is 20CB.
But I always receive an error message when I try a set command :
 
[16:01:24] INFO: Looping vclient...
[16:01:36] INFO: Sending command: [setBetriebArt]: WW
[72] Wed Jan 25 16:01:36 2023 : Client connected 127.0.0.1:48130 (FD:5)
[72] Wed Jan 25 16:01:36 2023 : Command: setBetriebArt WW
SRV ERR: command unknown
[72] Wed Jan 25 16:01:36 2023 : Command: quit
[72] Wed Jan 25 16:01:36 2023 : Process 72 released lock
[72] Wed Jan 25 16:01:36 2023 : Closed connection (fd:5)
server error
setBetriebArt WW:

Can you help me with this error

Regards

Nicolouts

**Comment by Razzer90** [comment]
Hey! There is a piece of code missing in the vito.xml file.
when you add the code you can run "set" and it changes the mode.
` <command name='setBetriebArtM1' protocmd='setaddr'>
            <addr>2301</addr>
            <len>1</len>
            <unit>BA</unit>
            <description>Setze Betriebsart M1</description>
            <device ID="2053">
                <addr>51</addr>
                <len>1</len>
                <unit>BA</unit>
            </device>
    </command>`

**Comment by nicolouts** [comment]
Hey,
Thank you for your feedback, but can you explain where I can find this file.
I have install the addon but I can't see structure folder.
I run HA in a virtual machine with hassio.
Regards
Nicolout

**Comment by Razzer90** [comment]
Yes, no problem, I installed the addon samba share. you can use it to access home assistant as a network drive. there you have the addons folder. I downloaded homeassistant-vcontrol as a zip, there you will find the vito.xml, there you change it accordingly. and put the unpacked and modified in the addons folder. then restart home assistant and the addon can then be found under Local add-ons and you can install it as normal. best greetings :)

**Comment by nicolouts** [comment]
Ok, thank you.
I have installed the addon by adding it to home assistant addon repository. So I don't have access to the main file.
I will follow your explanantion.
Regards
Nicolouts

**Comment by Razzer90** [comment]
https://github.com/Alexandre-io/homeassistant-vcontrol/pull/14/commits/ffd6a72ed2a6b838543f4151cf4f7a9830787844

:)

**Comment by raebbaer** [comment]
I have the same problem. I can't find the configuration file in order to add new commands. I have the latest HA installation (Home Assistant 2023.1.7) running on a raspberry. And I have installed the 1.8.2 addon from your repository. I use the "20CB" protocol and I am trying to set the partymode. The /etc/vcontrold/ directory is missing. And using the Samba addon I see only an emtpy addons directory. Can you please tell me, where I went wrong?

**Comment by Schm1tz1** [comment]
@raebbaer Same device, same issue. here. Looking into vito.xml of this project: https://github.com/Alexandre-io/homeassistant-vcontrol/blob/main/vcontrold/rootfs/etc/vcontrold/vito.xml and openv/vcontrold: https://github.com/openv/vcontrold/blob/master/xml/300/vito.xml you can see the it does not explicitly mention 20CB but only 2053 for this command. It is assuming some backwards compatibility of the protocols. That does not seem to apply here but I don't have any other sources. Maybe openv would be a good place for a Issue/FR. To add/fix this we probably either need a sniffer in a working setup or the Viessmann tools as described in https://github.com/sarnau/InsideViessmannVitosoft
